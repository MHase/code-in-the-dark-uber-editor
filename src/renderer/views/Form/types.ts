export interface FormValues {
  name: string;
  url: string;
}
