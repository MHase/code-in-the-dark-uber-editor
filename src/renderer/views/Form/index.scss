.Welcome {
  &__container {
    display: flex;
    align-items: center;
    justify-content: center;
    height: 100vh;
  }
}

.MuiButton-root.Welcome__btn {
  margin-top: 16px;
}
