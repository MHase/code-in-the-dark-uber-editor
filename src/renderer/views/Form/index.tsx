import React, { useCallback } from 'react';
import { Formik, Form } from 'formik';
import * as Yup from 'yup';
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';
import Container from '@material-ui/core/Container';

import { FormValues } from './types';

import './index.scss';

const SignupSchema = Yup.object().shape({
  name: Yup.string()
    .min(2, 'Too Short!')
    .max(50, 'Too Long!')
    .required('Required'),
  url: Yup.string()
    .url('Invalid ')
    .required('Required'),
});

const initialValues = {
  name: window.localStorage.getItem('name') || '',
  url: window.localStorage.getItem('url') || '',
};

const WelcomeForm = ({ ...props }) => {
  const handleSubmit = useCallback((values: FormValues, { setSubmitting }: { setSubmitting: Function }) => {
    window.localStorage.setItem('name', values.name);
    window.localStorage.setItem('url', values.url);
    setSubmitting(false);
    props.history.push('/editor');
  }, []);

  return (
    <React.Fragment>
      <Container maxWidth="sm" className="Welcome__container">
        <Formik initialValues={initialValues} validationSchema={SignupSchema} onSubmit={handleSubmit}>
          {({ values: { name, url }, errors, touched, isSubmitting, handleBlur, handleChange }) => {
            return (
              <Form>
                <TextField
                  type="text"
                  name="name"
                  label="Name"
                  value={name}
                  error={touched.name && Boolean(errors.name)}
                  helperText={touched.name && errors.name}
                  onChange={handleChange}
                  onBlur={handleBlur}
                  fullWidth
                  margin="normal"
                />
                <TextField
                  type="url"
                  name="url"
                  label="Url"
                  value={url}
                  error={touched.url && Boolean(errors.url)}
                  helperText={touched.url && errors.url}
                  onChange={handleChange}
                  onBlur={handleBlur}
                  fullWidth
                  margin="normal"
                />

                <Button
                  variant="contained"
                  color="secondary"
                  type="submit"
                  disabled={isSubmitting}
                  size="large"
                  className="Welcome__btn"
                >
                  Start
                </Button>
              </Form>
            );
          }}
        </Formik>
      </Container>
    </React.Fragment>
  );
};

export default WelcomeForm;
