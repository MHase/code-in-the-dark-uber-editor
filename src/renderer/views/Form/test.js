import React from 'react';
import { render } from '@testing-library/react';

import Form from './';
import StorageProvider from '../../context/storage';

describe('View | Form', () => {
  it('matches the snapshot', () => {
    const container = render(<Form />, { wrapper: StorageProvider });
    expect(container).toMatchSnapshot();
  });
});
