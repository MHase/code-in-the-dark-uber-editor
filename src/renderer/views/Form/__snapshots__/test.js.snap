// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View | Form matches the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiContainer-root Welcome__container MuiContainer-maxWidthSm"
      >
        <form>
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
              data-shrink="false"
            >
              Name
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input"
                name="name"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
              data-shrink="false"
            >
              Url
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input"
                name="url"
                type="url"
                value=""
              />
            </div>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root Welcome__btn MuiButton-contained MuiButton-containedSecondary MuiButton-sizeLarge"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              Start
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiContainer-root Welcome__container MuiContainer-maxWidthSm"
    >
      <form>
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink="false"
          >
            Name
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input"
              name="name"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink="false"
          >
            Url
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input"
              name="url"
              type="url"
              value=""
            />
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root Welcome__btn MuiButton-contained MuiButton-containedSecondary MuiButton-sizeLarge"
          tabindex="0"
          type="submit"
        >
          <span
            class="MuiButton-label"
          >
            Start
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
